infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -458,14 +483,18 @@ public CorfuStoreMetadata.Timestamp onClientListenerResume() {
         return preprocessAndGetTail();
     }
 
-    public UUID getOpaqueStreamToTrack(ReplicationSubscriber subscriber) {
-        switch(subscriber.getModel()) {
+    public UUID getOpaqueStreamToTrack(LogReplicationSession session) {
+        switch(session.getSubscriber().getModel()) {
             case FULL_TABLE:
                 return ObjectsView.getLogReplicatorStreamId();
             case LOGICAL_GROUPS:
-                return ObjectsView.getLogicalGroupStreamTagInfo(subscriber.getClientName()).getStreamId();
+                return ObjectsView.getLogicalGroupStreamTagInfo(session.getSubscriber().getClientName()).getStreamId();
+            case ROUTING_QUEUES:
+                String logEntrySyncStreamTag = ((LogReplicationRoutingQueueConfig) sessionToConfigMap.get(session))
```

Comment:
there will be a separate opaque stream to track if in snapshot sync.  We will now use the same approach to find entries to replicate in snapshot sync also.  It cannot be assumed that this method will get invoked in LogEntry Sync only